src/cls/queue/cls_queue_client.cc

```diff
@@ -0,0 +1,118 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+#include <errno.h>
+
+#include "cls/rgw/cls_rgw_ops.h"
+#include "cls/queue/cls_queue_const.h"
+#include "cls/queue/cls_queue_client.h"
+
+#include "common/debug.h"
+
+using namespace librados;
+
+void cls_rgw_gc_create_queue(ObjectWriteOperation& op, string& queue_name, uint64_t& size, uint64_t& num_urgent_data_entries)
+{
```

Comment:
@mattbenjamin @cbodley : Is it required to fix the number of urgent data entries? Or can we allow as many urgent data entries as needed (with spill over to xattrs)?. Also Matt suggested that for applications like GC that need urgent data, the queue start offset should be bigger than 1K, possibly 4K, so I am thinking that for apps that need urgent data the queue start offset will be set to 4K (which again will be configurable for testing purposes) and for apps that do not need urgent data, the queue start offset will be set to 1K. Is this ok?